File: src/main/java/infinitespire/relics/BrokenMirror.java
package infinitespire.relics;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.powers.ReflectionPower;
import com.megacrit.cardcrawl.relics.AbstractRelic;

import infinitespire.InfiniteSpire;
import infinitespire.abstracts.Relic;

public class BrokenMirror extends Relic {
	
	public static final String ID = InfiniteSpire.createID("Broken Mirror");
	
	public BrokenMirror() {
		super(ID, "brokenmirror", RelicTier.UNCOMMON, LandingSound.CLINK);
	}

	@Override
	public AbstractRelic makeCopy() {
		return new BrokenMirror();
	}

	@Override
	public void atBattleStart() {
		AbstractPlayer p = AbstractDungeon.player;
		AbstractDungeon.actionManager.addToTop(new ApplyPowerAction(p, p, new ReflectionPower(p, 3), 3));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/BrokenMirror.kt
package com.blanktheevil.infinitespire.relics

import basemod.AutoAdd
import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.megacrit.cardcrawl.powers.ReflectionPower

@AutoAdd.Ignore
class BrokenMirror : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "Broken Mirror".makeID()
    private const val IMG = "brokenmirror"
    private val TIER = RelicTier.UNCOMMON
    private val SOUND = LandingSound.CLINK
  }

  override fun atBattleStart() {
    player.applyPower(
      ReflectionPower(player, 3)
    )
  }
}